ntbs-service-unit-tests/DataMigration/ImportValidatorTest.cs

```diff
@@ -127,26 +127,463 @@ public async Task ImportValidatorValidatesCollectionModels()
+        [InlineData(null)]
+        [InlineData(1)]
+        public async Task ImportValidatorCleansContactTracingValuesBasedOn_ChildrenStartedTreatment(int? valueEntered)
+        {
+            // Arrange
+            var notification = CreateBasicNotification();
+
+            notification.ContactTracing = new ContactTracing
+            {
+                ChildrenIdentified = 3,
+                ChildrenScreened = 3,
+                ChildrenActiveTB = 1,
+                ChildrenLatentTB = 1,
+                ChildrenStartedTreatment = valueEntered,
+                ChildrenFinishedTreatment = 3,
+            };
+
+            // Act
+            var validationResults = await _importValidator.CleanAndValidateNotification(null, RunId, notification);
+
+            // Assert
+            // Validation errors
+            var errorMessages = validationResults.Select(r => r.ErrorMessage).ToList();
+            Assert.Empty(errorMessages);
+
+            // Data cleaning warnings
+            _importLoggerMock.Verify(s => s.LogNotificationWarning(null, RunId, notification.LegacyId,
+                ValidationMessages.ContactTracingChildrenFinishedTreatment));
+
+            Assert.Equal(3, notification.ContactTracing.ChildrenIdentified);
+            Assert.Equal(3, notification.ContactTracing.ChildrenScreened);
+            Assert.Equal(1, notification.ContactTracing.ChildrenActiveTB);
+            Assert.Equal(1, notification.ContactTracing.ChildrenLatentTB);
+            Assert.Equal(valueEntered, notification.ContactTracing.ChildrenStartedTreatment);
+            Assert.Null(notification.ContactTracing.ChildrenFinishedTreatment);
+        }
+
+        [Fact]
+        public async Task ImportValidatorCleansContactTracingValues_WhenTooManyOfEachAdultsCaseType()
+        {
+            // Arrange
+            var notification = CreateBasicNotification();
+
+            notification.ContactTracing = new ContactTracing
+            {
+                AdultsIdentified = 3,
+                AdultsScreened = 3,
+                AdultsActiveTB = 2,
+                AdultsLatentTB = 2,
+                AdultsStartedTreatment = 1,
+                AdultsFinishedTreatment = 1,
+            };
+
+            // Act
+            var validationResults = await _importValidator.CleanAndValidateNotification(null, RunId, notification);
+
+            // Assert
+            // Validation errors
+            var errorMessages = validationResults.Select(r => r.ErrorMessage).ToList();
+            Assert.Empty(errorMessages);
+
+            // Data cleaning warnings
+            _importLoggerMock.Verify(s => s.LogNotificationWarning(null, RunId, notification.LegacyId,
+                ValidationMessages.ContactTracingAdultsActiveTB));
+
+            Assert.Equal(3, notification.ContactTracing.AdultsIdentified);
+            Assert.Equal(3, notification.ContactTracing.AdultsScreened);
+            Assert.Null(notification.ContactTracing.AdultsActiveTB);
```

Comment:
This is surprising - I would have expected that where we have a choice, we'd choose to keep the "Active" data rather than the "Latent" data. I don't know if this is likely to actually affect any records... if it is then it's worth running by Nancy.

Reply:
So my reasoning here was that other values (treatment started and finished) depend on the latent value, but no values depend on the active value, so if we remove the latent value we definitely need to remove two more pieces of data if they exist, but if we remove the active one then we might be able to keep the rest.

One potential misfeature of this is that this will transform 
```
Adults Screened: 3
Adults Active TB: 1
Adults Latent TB: 4
```
into
```
Adults Screened: 3
Adults Active TB: null
Adults Latent TB: null
```
when if we chose to remove the latent TB value then we could have left the active one. I'm not sure how many edge cases we want to bend over backwards to cover though, given we know the data to be bad if we've gotten this far into the cleaning process

Reply:
I see what you're saying... I think we really need some data on how many records this will affect. It may be that it just doesn't really matter.

Reply:
I've spoken to Nancy and she said we should remove all subsequent contact tracing counts, both active and latent, if the counts don't add up to the screening number, and I've just implemented that

Reply:
Ok, sounds good